### DISCUSSION TITLE
Is it possible to monitor keyboard layout change?

### DISCUSSION THREAD

**Discussion by vitaly-zdanevich** [jezek/xgb#28] [Q&A]
Thanks.

**Comment by jezek** [comment]
Hi, I did some search and [consultation](https://chatgpt.com/share/66fc5400-bf38-8012-a4ef-10935364ac4a)*. I'm sorry, but the answer is *currently no*.

When using `C` and `xcb`, there is an extension `xkb`, which can [track keyboard events](https://www.x.org/releases/X11R7.7/doc/libX11/XKB/xkblib.html#Xkb_Events).

However, I didn't find the `xkb` extension in this repo. After digging deeper into this project I found this [note from previous maintainer](https://github.com/jezek/xgb/blob/master/doc.go#L140):
```
XKB is the only extension that intentionally does not work, although I suspect
that GLX also does not work (however, there is Go source code for GLX that
compiles, unlike XKB). I don't currently have any intention of getting XKB
working, due to its complexity and my current mental incapacity to test it.
```

So, I'm afraid, if you want to monitor keyboard layout changes using `xgb`, you'll have to do some work and make it functional.

I'm sorry, I couldn't help more.

Note: Most of the code in the sub-directories is generated using `xgbgen`. In `Makefile` you can see how. Maybe you can try to generate the `xkb` extension and than see, what works and what doesn't.

*) The examples in the consultation above may not be correct, but inspiration can be drawn from them.
